### PR TITLE
Merge in xenops.git

### PR DESCRIPTION
**Pull request by robhoes**
This merges in xenops.git. The repo is first added into the directory `xenops/`, while preserving its history (hence the large number of commits under this PR). This was done with the following command:

    git subtree add --prefix xenops git@github.com:xapi-project/xenops master

All we were actually using from xenops.git is the `list_domains` program. The files belonging to it are moved to `list_domains/`, updated to fit into the local build, and the remnants of xenops removed.

### CODE REVIEW

File: list_domains/dune [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(executable
+  (name list_domains)
+  (public_name list_domains)
+  (package xapi-xenopsd-xc)
+  (flags (:standard -bin-annot))
```
Comment (edwintorok) [inline list_domains/dune]: I don't think we need `bin-annot` here specifically, `merlin` works fairly well with the defaults.

File: list_domains/table.ml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  let column x = List.map (fun row -> List.nth row x) rows in
+  let cols = List.map column (mkints numcols) in
+
+  let max n str = max n (String.length str) in
+  List.map (List.fold_left max 0) cols
+
+let print (rows: string list list) = match rows with
+	| [] -> ()
+	| _ ->
+		let widths = compute_col_widths rows in
```
Comment (edwintorok) [inline list_domains/table.ml]: The indentation here is quite wide, did you use ocp-indent?

File: list_domains/dune (hunk 1 above)
Comment (robhoes) [reply list_domains/dune]: I'll remove that line.

File: list_domains/table.ml (hunk 2 above)
Comment (robhoes) [reply list_domains/table.ml]: Turns out it's tabs. I'll reindent it.

**Pull request merged**